frontend/src/metabase/visualizations/echarts/cartesian/model/series.ts

```diff
@@ -0,0 +1,150 @@
+import type {
+  SingleSeries,
+  DatasetData,
+  RowValue,
+  DatasetColumn,
+  RawSeries,
+} from "metabase-types/api";
+import type { CartesianChartColumns } from "metabase/visualizations/lib/graph/columns";
+import type {
+  SeriesModel,
+  VizSettingsKey,
+} from "metabase/visualizations/echarts/cartesian/model/types";
+import { getDatasetKey } from "metabase/visualizations/echarts/cartesian/model/dataset";
+import type {
+  Formatter,
+  RenderingContext,
+} from "metabase/visualizations/types";
+
+type MetricSeriesParams = {
+  metricColumn: DatasetColumn;
+};
+
+type BreakoutSeriesParams = {
+  breakoutColumn: DatasetColumn;
+  breakoutValue: RowValue;
+  formatValue: Formatter;
+};
+
+type SeriesVizSettingsKeyParams = {
+  cardName?: string;
+  isFirstCard: boolean;
+} & (MetricSeriesParams | BreakoutSeriesParams);
+
+export const getSeriesVizSettingsKey = ({
```

Comment:
```suggestion
 const getSeriesVizSettingsKey = ({
```
We should remove `export` so the reader knows this just a helper for this file, not a top level function called somewhere else, unless we plan on doing so in a later PR.